haystack/nodes/reader/base.py

```diff
@@ -113,6 +140,14 @@ def run_batch(  # type: ignore
         add_isolated_node_eval: bool = False,
     ):
         self.query_count += len(queries)
+
+        # Remove empty documents before making predictions
+        if len(documents) > 0:
+            if isinstance(documents[0], Document):
+                documents = [d for d in documents if d.content.strip() != ""]  # type: ignore[union-attr, assignment]
+            else:
+                documents = [[d for d in docs_per_query if d.content.strip() != ""] for docs_per_query in documents]  # type: ignore[union-attr]
```

Comment:
I changed the code so that we now account for Documents of type table. 👍 The `#type: ignore statements` are still needed because mypy is unsure whether `for docs_per_query in documents` enumerates documents or lists of documents and complains about that.